src/sentry/static/sentry/app/views/usageStats/usageChart/index.tsx

```diff
@@ -0,0 +1,337 @@
+import React from 'react';
+import Color from 'color';
+import {EChartOption} from 'echarts';
+import {withTheme} from 'emotion-theming';
+
+import BaseChart from 'app/components/charts/baseChart';
+import Legend from 'app/components/charts/components/legend';
+import Tooltip from 'app/components/charts/components/tooltip';
+import xAxis from 'app/components/charts/components/xAxis';
+import barSeries from 'app/components/charts/series/barSeries';
+import {ChartContainer, HeaderTitleLegend} from 'app/components/charts/styles';
+import Panel from 'app/components/panels/panel';
+import ChartPalette from 'app/constants/chartPalette';
+import {t} from 'app/locale';
+import {DataCategory, SelectValue} from 'app/types';
+import {formatAbbreviatedNumber} from 'app/utils/formatters';
+import commonTheme, {Theme} from 'app/utils/theme';
+
+import {formatUsageWithUnits, GIGABYTE} from '../utils';
+
+import {getDateRange, getTooltipFormatter} from './utils';
+
+const COLOR_ERRORS = ChartPalette[4][3];
+const COLOR_ERRORS_DROPPED = Color(COLOR_ERRORS).lighten(0.25).string();
+
+const COLOR_TRANSACTIONS = ChartPalette[4][2];
+const COLOR_TRANSACTIONS_DROPPED = Color(COLOR_TRANSACTIONS).lighten(0.25).string();
+
+const COLOR_ATTACHMENTS = ChartPalette[4][1];
+const COLOR_ATTACHMENTS_DROPPED = Color(COLOR_ATTACHMENTS).lighten(0.5).string();
+const COLOR_PROJECTED = commonTheme.gray200;
+
+export const CHART_OPTIONS_DATACATEGORY: SelectValue<DataCategory>[] = [
+  {
+    label: 'Errors',
+    value: DataCategory.ERRORS,
+    disabled: false,
+  },
+  {
+    label: 'Transactions',
+    value: DataCategory.TRANSACTIONS,
+    disabled: false,
+  },
+  {
+    label: 'Attachments',
+    value: DataCategory.ATTACHMENTS,
+    disabled: false,
+  },
+];
+
+export enum ChartDataTransform {
+  CUMULATIVE = 'cumulative',
+  DAILY = 'daily',
+}
+
+export const CHART_OPTIONS_DATA_TRANSFORM: SelectValue<ChartDataTransform>[] = [
+  {
+    label: 'Cumulative',
+    value: ChartDataTransform.CUMULATIVE,
+    disabled: false,
+  },
+  {
+    label: 'Day-to-Day',
+    value: ChartDataTransform.DAILY,
+    disabled: false,
+  },
+];
+
+export enum SeriesTypes {
+  ACCEPTED = 'Accepted',
+  DROPPED = 'Dropped',
+  PROJECTED = 'Projected',
+}
+
+type Props = {
+  theme: Theme;
+
+  title?: React.ReactNode;
+  footer?: React.ReactNode;
```

Comment:
I like this separation of footer and passing the active selection state in as props :clap: